var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -275,49 +284,48 @@ def configure_args(self):
         configure_args.extend(self.enable_or_disable("debug"))
 
         if spec.satisfies("+cuda"):
+            configure_args.append(f"--with-cuda-home={spec['cuda'].prefix}")
             configure_args.extend(["--with-cuda", "--enable-curand", "--enable-cusparse"])
             cuda_arch_vals = spec.variants["cuda_arch"].value
             if cuda_arch_vals:
                 cuda_arch_sorted = list(sorted(cuda_arch_vals, reverse=True))
                 cuda_arch = cuda_arch_sorted[0]
                 configure_args.append(f"--with-gpu-arch={cuda_arch}")
             # New in 2.21.0: replaces --enable-cub
-            if spec.satisfies("@2.21.0:"):
+            if spec.satisfies("@2.21.0: ~umpire"):
                 configure_args.append("--enable-device-memory-pool")
-                configure_args.append(f"--with-cuda-home={spec['cuda'].prefix}")
-            else:
+            elif spec.satisfies("@:2.20.99"):
                 configure_args.append("--enable-cub")
-            if spec.satisfies("+cublas"):
-                configure_args.append("--enable-cublas")
+            if spec.satisfies("@2.29.0:"):
+                configure_args.extend(["--enable-cublas", "--enable-cusolver"])
         else:
             configure_args.extend(["--without-cuda", "--disable-curand", "--disable-cusparse"])
             if spec.satisfies("@:2.20.99"):
                 configure_args.append("--disable-cub")
+            if spec.satisfies("@2.29:"):
+                configure_args.append("--disable-cusolver")
 
         if spec.satisfies("+rocm"):
-            rocm_pkgs = ["rocsparse", "rocthrust", "rocprim", "rocrand"]
+            configure_args.append("--with-hip")
+            rocm_pkgs = ["rocthrust", "rocprim", "rocrand", "rocsparse", "rocblas"]
             if spec.satisfies("+superlu-dist"):
                 rocm_pkgs.append("hipblas")
-            rocm_inc = ""
-            for pkg in rocm_pkgs:
-                rocm_inc += spec[pkg].headers.include_flags + " "
+            if spec.satisfies("@2.29.0:"):
+                rocm_pkgs.extend(["rocblas", "rocsolver"])
```

Comment:
There is a bit of inconsistency here:
- rocblas is added twice - earlier at line:310 - perhaps that should be removed?
- here rocblas, rocsolver are conditional on '@2.29.0:'. Should `depends_on()` also have a similar conditional?

Reply:
Thank you for catching this, Satish! Fixed in [870a463](https://github.com/spack/spack/pull/49217/commits/870a463ee51fff0df662834dc1cc1744db6eee9f)